doc/src/bond_over_grpc.md

```diff
@@ -90,21 +96,141 @@ On the client side, the proxy stub establishes a connection to the server like t
+  for each of the methods defined in the service IDL, serving as a base for the
+  concrete implementation which will provide the actual server-side business
+  logic.
+* The proxy stub, which is an inner class named `Client` (e.g.:
+  `Example::Client`). [^2] This is used to invoke the service from the
+  client side.
+
+To build the service functionality, simply write a concrete service
+implementation by subclassing the server base and supplying the business logic:
+
+    class ExampleServiceImpl final : Example::Service
+    {
+        void ExampleMethod(
+            bond::ext::gRPC::unary_call<
+                bond::bonded<ExampleRequest>,
+                bond::bonded<ExampleResponse>> call) override
+        {
+            ExampleRequest request = call.request().Deserialize();
+            ExampleResponse reply;
+
+            // Service business logic goes here
+
+            call.Finish(bond::bonded<ExampleResponse >{response}, Status::OK);
+        }
+    }
+
+This service implementation is hooked up to a gRPC server as follows:
+
+    auto ioManager = std::make_shared<bond::ext::gRPC::io_manager>();
+    auto threadPool = std::make_shared<bond::ext::gRPC::thread_pool>();
+    const std::string server_address(Host + ":" + Port);
+
+    ExampleServiceImpl service;
+    bond::ext::gRPC::server_builder builder;
+    builder.SetThreadPool(threadPool);
+    const std::string server_address(Host + ":" + Port);
+    builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+    builder.RegisterService(&service);
+    std::unique_ptr<bond::ext::gRPC::server> server(builder.BuildAndStart());
+
+At this point the server is ready to receive requests and route them to the
+service implemetation.
+
+On the client side, the proxy stub establishes a connection to the server like this:
+
+    auto ioManager = std::make_shared<bond::ext::gRPC::io_manager>();
+    auto threadPool = std::make_shared<bond::ext::gRPC::thread_pool>();
+    const std::string server_address(Host + ":" + Port);
+
+    Example::Client client(
+        grpc::CreateChannel(server_address, grpc::InsecureChannelCredentials()),
+        ioManager,
+        threadPool);
+
+The proxy stub can then be used to make calls to the server as follows:
+
+    ExampleRequest request;
+    // Fill in request fields here
+
+    bond::ext::gRPC::wait_callback<ExampleResponse> cb;
+    client.AsyncExampleMethod(&context, request, callback);
+
+    callback.wait();
+    ExampleResponse response = callback.response().Deserialize();
+    // Examine response here
+
+Note these APIs are significantly different from the APIs presented in the
+gRPC documentation; Bond-over-gRPC is attempting to provide a more
+straightforward API for aynchronous communication than gRPC currently presents
+in C++. The use of `wait_callback<T>` here is for illustrative purposes; any
+callback implementation with the same signature can be used. Note also the use
```

Comment:
maybe we should mention that the callback signature should receive the input param type and grpc::Status returning void

`std::function<void(const ::bond::bonded< ::helloworld::HelloReply>&, const ::grpc::Status&)>`